scripts/build-binary.ps1

```
@@ -0,0 +1,59 @@
+$PyVersion = & $Python -c "import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')"
+Write-Host "[build] Using Python $PyVersion"
+
+$PyMajor = & $Python -c "import sys; print(sys.version_info.major)"
+$PyMinor = & $Python -c "import sys; print(sys.version_info.minor)"
+if ([int]$PyMajor -lt 3 -or ([int]$PyMajor -eq 3 -and [int]$PyMinor -lt 10)) {
+    Write-Error "[build] ERROR: Python >= 3.10 required, got $PyVersion"
+    exit 1
+}
+
+# Create isolated venv
+Write-Host "[build] Creating virtual environment at $VenvDir"
```

PSScriptAnalyzer (1.23.0)

[warning] File 'build-binary.ps1' uses Write-Host. Avoid using Write-Host because it might not work in all hosts, does not work when there is no host, and (prior to PS 5.0) cannot be suppressed, captured, or redirected. Instead, use Write-Output, Write-Verbose, or Write-Information.

(PSAvoidUsingWriteHost)